client/src/controllers/ZoneController.test.js

```diff
@@ -166,4 +166,110 @@ describe('ZoneController', () => {
       expect(element.classList.contains('hovered')).toBe(false);
     });
   });
+
```

Comment:
Would you be able to add a unit test for the setting of the mode-value based on the initial class?

To do this, I recommend reading that the `data-w-zone-value` after awaiting setup for the html with the appropriate class. 

While this is kind of digging into the internals a bit it's still ok and is the simplest way to test this works. 

Please add two tests, one for active classes & one for inactive. 

Thankyou. This is incredible stuff and it's looking really good.

Reply:
Sure, I will add it.